crates/bevy_transform/src/transform_propagate_system.rs

```diff
@@ -2,75 +2,98 @@ use crate::components::{Children, GlobalTransform, Parent, Transform};
 use bevy_ecs::{
     entity::Entity,
     query::{Changed, With, Without},
-    system::Query,
+    system::{Local, Query},
 };
 
+/// Used for [`transform_propagate_system`]. Ignore otherwise.
+pub(crate) struct Pending {
+    parent: *const GlobalTransform,
+    changed: bool,
+    child: Entity,
+}
+
+// SAFE: Values are cleared after every frame and cannot otherwise be
+// constructed without transmute.
+unsafe impl Send for Pending {}
+// SAFE: Values are cleared after every frame and cannot otherwise be
+// constructed without transmute.
+unsafe impl Sync for Pending {}
+
+/// Update [`GlobalTransform`] component of entities based on entity hierarchy and
+/// [`Transform`] component.
+pub(crate) fn transform_propagate_flat_system(
+    mut root_query: Query<
+        (&Transform, &mut GlobalTransform),
+        (Without<Parent>, Without<Children>, Changed<Transform>),
+    >,
+) {
+    for (transform, mut global_transform) in root_query.iter_mut() {
+        *global_transform = GlobalTransform::from(*transform);
+    }
+}
+
 /// Update [`GlobalTransform`] component of entities based on entity hierarchy and
 /// [`Transform`] component.
-pub fn transform_propagate_system(
+pub(crate) fn transform_propagate_system(
```

Comment:
Is this `pub(crate)` due to the visibility of `Pending`?

I think we might be able to just expose `Pending` and just make it `#[doc(hidden)]`. It's not like it has any public interface.

Reply:
That's part of it. The other half is primarily to avoid confusion as there are now two different propagation systems. It may be better to encourage people to add the full TransformPlugin instead of just individual systems. If it's really desirable to just add the transform propagation system, we should have a public API that adds both of them to a Schedule.